server/test/endpoint_admin_personas_roles.test.js

```diff
@@ -70,6 +84,71 @@ describe('Admin Personas Roles', () => {
     })
   })
 
+  describe('POST:/personas/{persona_key}/roles', () => {
+    const postPersonasRoles = async (requestPayload, header = requestHeaders, personaKey = testPersonaKey) => {
+      return await fetch(`${url}/personas/${personaKey}/roles`, {
+        method: 'POST',
+        headers: header,
+        body: JSON.stringify(requestPayload)
+      })
+    }
+
+    it('should return 201, empty response body, and Location header to get all the persona roles', async () => {
+      const response = await postPersonasRoles(validTenantRoles.map(({ key }) => key))
+      assert.equal(response.status, 201)
+      assert.equal(response.headers.get('Location'), response.url)
+      const responseBody = await response.text()
+      assert.equal(responseBody, '')
+    })
+
+    it('should return 400, a role from a client which does not belong to the same tenant cannot be assigned to persona', async () => {
+      const response = await postPersonasRoles([...validTenantRoles, invalidPersona].map(({ key }) => key))
+      assert.equal(response.status, 400)
+    })
+
+    it('should return 400, for four different invalid request payloads', async () => {
+      const [emptyBodyResponse, invalidBodyResponse, invalidRoleResponse, nonUniqueRolesResponse] = await Promise.all(
+        [
+          postPersonasRoles(),
+          postPersonasRoles({}),
+          postPersonasRoles([0]),
+          postPersonasRoles([1, 1])
+        ]
+      )
+      assert.equal([
+        emptyBodyResponse.status,
+        invalidBodyResponse.status,
+        invalidRoleResponse.status,
+        nonUniqueRolesResponse.status].every((status) => status === 400), true)
+    })
+
+    it('should return 401, unauthorized token', async () => {
+      const userAccessToken = await getTestUser1IdPToken()
+      const response = await postPersonasRoles(
+        [validRoleKey],
+        {
+          ...requestHeaders,
+          Authorization: `Bearer ${userAccessToken}`
+        })
+      assert.equal(response.status, 401)
+    })
+
+    it('should return 404, fail to create persona roles for an invalid persona', async () => {
+      const response = await postPersonasRoles([validRoleKey], requestHeaders, invalidPersona)
+      assert.equal(response.status, 404)
+    })
+
+    it('should return 409, fail to create persona roles due to duplication', async () => {
```

Comment:
@k1ch same question as above on if the API should work if the request contains duplicate roles? 
Also, similar question on if the request contains a role that a user is already in. Should that error or "work" as the end result is that the user is in the given role?

Reply:
If we do not want to enforce role key uniqueness on body request validation, we can easily turn the Array into a Set and obtain the unique role keys. 
Regarding the `409` error response, if we decide to ignore duplicated roles, we will have the scenario as I mentioned in the other comment:
>Also, regarding API behavior, imagine the body request is [ 5 ], and the user already has this role. If that is ignored, we will respond with a 201 status code, which I can argue is a confusing behavior. Having said that, it is more about our preference.